### PR TITLE
Fixed invalid phys obj

### PR DESCRIPTION
**Pull request by mcNuggets1**
If phys obj is invalid, don't wake or disable motion.

### CODE REVIEW

File: entities/entities/spawned_weapon/init.lua [hunk 1]
```diff
@@ -14,10 +14,10 @@ function ENT:Initialize()
         self:SetModel("models/weapons/w_rif_ak47.mdl")
         self:PhysicsInit(SOLID_VPHYSICS)
         phys = self:GetPhysicsObject()
```
Comment (Kefta) [inline entities/entities/spawned_weapon/init.lua]: You should add another phys:Wake() here

Comment (mcNuggets1) [reply entities/entities/spawned_weapon/init.lua]: Oh yeah.
Thanks for letting me know.

**Comment by mcNuggets1** [comment]
Should be fine now.

**Comment by Bo98** [comment]
What scenarios does this happen apart from not having CS:S mounted?

**Comment by mcNuggets1** [comment]
Invalid food, money.

I just created this pullrequest, because if css is not mounted, there is no need to spam the console with errors, it already tells every player about css not being mounted.

**Comment by Bo98** [comment]
Ok, I was just making sure we weren't silencing a different problem.

**Comment by Kefta** [comment]
Physics objects can also be failed to created if there's too many in the physenv just like entities.

**Comment by FPtje** [comment]
Is there a function to check whether the limit is reached? I'd rather use that than fail silently. With this people are staring at stuff that floats around with no message in console explaining what's going on.

Like we did with the entity limit. Throw a big error explaining what's going on.

**Comment by Kefta** [comment]
Nope, nor does the engine have one. The public vphysics API is extremely limited, but hitting the limit is extremely rare.

**Comment by Kefta** [comment]
You'll almost always hit the entity limit before the physobj assuming your entities have 1-3 physobjs each.

**Comment by FPtje** [comment]
So the expected cause for this error is something on the side of the user, be it CSS not being installed, or a model being set that doesn't exist or doesn't have a physics object.

For CSS not being mounted I understand that the error is redundant. For the other case, though, I think the error is *very* relevant for the user. In that case removing the error would make it more difficult for the user to find out what's going on. IIRC the error isn't that informative in the first place, perhaps we can throw a nicer error that explains the situation better?

The goal here is make it as easy as possible for server owners/developers to solve their own problem, instead of posting on the DarkRP forums about their stuff floating about or errors they don't understand.

**Comment by mcNuggets1** [comment]
There is an error for CSS already.
I never saw it happen, that the physics entity limit was hit.

**Comment by Kefta** [comment]
"Tried to use NULL physobj" also isn't a helpful error. I would change all PhysicsInit calls to something along the lines of:

```
if (not self:PhysicsInit(SOLID_VPHYSICS)) then
   print(string.format("Failed to create the physics of %s, missing model or hit physics limit", self:GetModel() or "")
end
```
``print`` is a bit silent but throwing a lua would be obnoxious on servers with missing models pretty quickly.

**Comment by FPtje** [comment]
Yeah, that's what I had in mind. Some explanation like that.

About print vs error, a missing model renders the entity entirely unusable. It completely breaks it. It's an error state, error states warrant errors to be thrown. I'd use the `DarkRP.error` mechanism, which pesters any joining admin to fix the error. DarkRP should tell them about the problem before they find themselves staring at some floating entity.

**Comment by Kefta** [comment]
I would use ``DarkRP.errornoHalt`` at the very least to let the entity still continue it's creation even without physics.

**Comment by FPtje** [comment]
Yeah, that's the same nagging mechanism. Not halting does mean there could be other errors occurring during initialisation.

**Comment by Kefta** [comment]
That is exactly what this PR fixes - failing to create physics will not create additional Lua errors.

**Comment by FPtje** [comment]
@mcNuggets1 The commit messages in this PR are descriptive. I can tell what they do from reading them. Thanks.

**Comment by FPtje** [comment]
The PR is merged, also with the following change:

`self:PhysicsInit(SOLID_VPHYSICS)` -> `DarkRP.ValidatedPhysicsInit(self, SOLID_VPHYSICS)`

That one throws an error on failure. I've refrained from overriding `PhysicsInit` entirely, since addons might want their own checks and recover in their own way. Overriding it and always throw an error on failure would diminish the return bool of the function.

**Comment by mcNuggets1** [comment]
I think we still check for isvalid in some entities for the physics object.

**Pull request merged**